test/extended/node_tuning/node_tuning.go

```diff
@@ -111,6 +117,81 @@ var _ = g.Describe("[sig-node-tuning] NTO should", func() {
 
 		err = fmt.Errorf("case: %v\nexpected error got because of %v", g.CurrentSpecReport().FullText(), fmt.Sprintf("stalld service restarted : %v", errWait))
 		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	//OCPBUGS-18052
+	g.It("SNO installation does not finish due to wait for non-existing machine-config [Early]", func() {
+		isSNO := isSNOCluster(oc)
+		if !isSNO {
+			g.Skip("only test on SNO cluster, skipping it ...")
+		}
+
+		var (
+			mcpConfigDaemonset *corev1.Pod
+			errMasterUpdating  error
+		)
+
+		ctx := context.TODO()
+		nodeClient := oc.KubeClient().CoreV1().Nodes()
+		firstMasterNodeName, err := getFirstMasterNode(ctx, nodeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("assert the status of mcp master")
+
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+		dynamicClient := dynamic.NewForConfigOrDie(config)
+		//Wait for 10 minutes and check mcp master stay on normal state
+		errWait := wait.Poll(30*time.Second, 600*time.Second, func() (bool, error) {
```

Comment:
10*time.Minute would be better to read IMO

Reply:
Fixed in new code